### PR TITLE
Z algorithm in C++

### PR DESCRIPTION
**Pull request by SangeetaMishra**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#417

### CODE REVIEW

File: C-Plus-Plus/cp/zalgorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                z[i] = right-left;
+                right--;
+            }
+        }
+    }
+}
+
+int main(){
+    string text, pattern;
+    cin >> text >> pattern;
```
Comment (atarax665) [inline C-Plus-Plus/cp/zalgorithm.cpp]: Prompt user to input the original string and the pattern to be searched separately.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Format your code using https://www.techiedelight.com/tools/clike

File: C-Plus-Plus/cp/zalgorithm.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            cout << "Pattern found at index " << i - m - 1 << "\n";
+        }
+    }
+    return 0;
+}
+
+/*
+Time Complexity: O(length of text + length of pattern)
+
+Input
```
Comment (atarax665) [inline C-Plus-Plus/cp/zalgorithm.cpp:62]: Format the input according to the changes you have made recently.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by iamrajiv — APPROVED** [review]

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Review by atarax665 — APPROVED** [review]

**Pull request merged**